llvm/test/CodeGen/AArch64/machine-licm-hoist-load.ll

```diff
@@ -0,0 +1,428 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 3
+; RUN: llc -mtriple=aarch64-linux-gnu < %s | FileCheck %s
+
+define i64 @one_dimensional(ptr %a, ptr %b, i64 %N, i64 %M, i64 %K) {
```

Comment:
It would also be nice to have a version of this with memcmp of 5 or 6 bytes to show that both load (+ rev + or) instructions get hoisted out.